### ISSUE TITLE
Ctrl-C quitting julia

### ISSUE THREAD

**Issue by fieker** [oscar-system/Polymake.jl#321]
julia> using Polymake
[ Info: Precompiling Polymake [d720cf60-89b5-51f5-aff5-213f193123e7]
[ Info: Generating module common
[ Info: Generating module ideal
[ Info: Generating module graph
[ Info: Generating module fulton
[ Info: Generating module fan
[ Info: Generating module group
[ Info: Generating module polytope
[ Info: Generating module topaz
[ Info: Generating module tropical
[ Info: Generating module matroid
polymake version 4.1
Copyright (c) 1997-2020
Ewgenij Gawrilow, Michael Joswig, and the polymake team
Technische Universität Berlin, Germany
https://polymake.org

This is free software licensed under GPL; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

julia> s = 0; for i=1:100000 global s+= sin(BigFloat(i))*tan(i^2); end
^C
[fieker@pitti src]$ 

ctrl-c is properly caught in the prompt, but when programms are running, it kills julia (the example is any julia command with a non-trivial runtime as I need time to hit ctrl-c after enter)

**Comment by benlorenz** [comment]
We had a patch for this in place but it somehow got lost in the transition to artifacts, I have added it again for the 4.2 bump of `polymake_jll` (https://github.com/JuliaPackaging/Yggdrasil/pull/1695). In the meantime you can apply [this small patch](https://github.com/benlorenz/Yggdrasil/blob/polymake/P/polymake/bundled/patches/sigint.patch) manually to the files in `polymake_jll.artifact_dir`.

**Comment by fieker** [comment]
On Thu, Sep 24, 2020 at 02:10:31AM -0700, Benjamin Lorenz wrote:
> We had a patch for this in place but it somehow got lost in the transition to artifacts, I have added it again for the 4.2 bump of `polymake_jll` (https://github.com/JuliaPackaging/Yggdrasil/pull/1695). In the meantime you can apply [this small patch](https://github.com/benlorenz/Yggdrasil/blob/polymake/P/polymake/bundled/patches/sigint.patch) manually to the files in `polymake_jll.artifact_dir`.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/oscar-system/Polymake.jl/issues/321#issuecomment-698220155

Thanks!
 Claus

**Comment by fingolfin** [comment]
@benlorenz's PR was merged into Yggdrasil as was https://github.com/JuliaRegistries/General/pull/22662 so now one can test this directly.

Now I guess the next step is to adjust Polymake.jl's `Project.toml` to be compatible with polymake_jll 4.2.0 (right now it says `~4.1.0` so it refuses to load with the newer polymake_jll) ?  Ah, but I just tried that, and it didn't work (on macOS), because `libpolymake_julia` was linked against polymake_jll 4.2.0...

```
[ Info: Precompiling Polymake [d720cf60-89b5-51f5-aff5-213f193123e7]
ERROR: LoadError: InitError: could not load library "/Users/mhorn/.julia/artifacts/a8b6c019daa49e71c8c51cea833252241d5511fb/lib/libpolymake_julia.dylib"
dlopen(/Users/mhorn/.julia/artifacts/a8b6c019daa49e71c8c51cea833252241d5511fb/lib/libpolymake_julia.dylib, 1): Library not loaded: @rpath/libpolymake.4.1.dylib
  Referenced from: /Users/mhorn/.julia/artifacts/a8b6c019daa49e71c8c51cea833252241d5511fb/lib/libpolymake_julia.dylib
  Reason: image not found
...
```

**Comment by benlorenz** [comment]
I am preparing a pull request to rebuild + bump libpolymake_julia.

**Comment by benlorenz** [comment]
This should be fixed since 0.5.1.
